### DISCUSSION TITLE
How to generate a monochrome bitmap with a bit depth of 1 in BMP format?

### DISCUSSION THREAD

**Discussion by jingxin2015** [dlemstra/Magick.NET#1619] [Help] [answered]
Due to printing requirements, I need to use it NET6 generates a monochrome bitmap in BMP format with a bit depth of 1, a white background, and black text. May I ask Magick Can NET be implemented? My program needs to be deployed on the Ubuntu system, thank you!

**Comment by dlemstra** [comment]
This should already be supported? What have you tried so far?

**Comment by jingxin2015** [comment]
@dlemstra 

             var fontPath = webHostEnvironment.WebRootPath + "/fonts/SimSun.ttf".Replace('/', Path.DirectorySeparatorChar);

            using var image = new MagickImage(MagickColors.White, 500, 700);
            image.Format = MagickFormat.Bmp;
            image.Depth = 1;

            var drawables = new Drawables();

            drawables.Font(fontPath, FontWeight.Bold, FontStretch.Normal);
            drawables.TextEncoding(Encoding.UTF8);
            drawables.FontPointSize(12);
            drawables.FillColor(MagickColors.Black);
            drawables.Text(0, 0, "单色位图");

            drawables.Draw(image);

            image.Write(webHostEnvironment.WebRootPath + "/images/test1bit.bmp".Replace('/', Path.DirectorySeparatorChar));

I tried the above code and the generated image is still 32-bit deep. If the format is set to MagickFormat Bmp3 generates a depth of 24 bits.

**Reply by dlemstra** [reply, marked as answer]
You should set `image.ColorType` to `ColorType.Bilevel` after you have called `.Draw(image)`. This makes sure that the image becomes monochrome.

**Reply by jingxin2015** [reply]
You're right, thank you very much!
